android/modules/gesture/src/java/ti/modules/titanium/gesture/GestureModule.java

```diff
@@ -178,38 +178,31 @@ public void onSensorChanged(SensorEvent event)
 	@Kroll.getProperty @Kroll.method
 	public boolean isPortrait()
 	{
-		// Deprecated in 6.1.0 in parity-favor of Ti.Gesture.portrait
-		return TiApplication.getInstance().getResources().getConfiguration().orientation == Configuration.ORIENTATION_PORTRAIT;
+		return TiOrientationHelper.isPortrait;
 	}
 
 	@Kroll.getProperty @Kroll.method
 	public boolean isLandscape()
 	{
-		// Deprecated in 6.1.0 in parity-favor of Ti.Gesture.landscape
-		return TiApplication.getInstance().getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE;
+		return !TiOrientationHelper.isPortrait;
 	}
 		
 	@Kroll.getProperty @Kroll.method
 	public boolean getPortrait()
 	{
-		return TiApplication.getInstance().getResources().getConfiguration().orientation == Configuration.ORIENTATION_PORTRAIT;
+		return TiOrientationHelper.isPortrait;
 	}
 
 	@Kroll.getProperty @Kroll.method
 	public boolean getLandscape()
 	{
-		return TiApplication.getInstance().getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE;
+		return !TiOrientationHelper.isPortrait;
 	}
 
 	@Kroll.getProperty @Kroll.method
 	public int getOrientation()
 	{
-	    DisplayMetrics dm = new DisplayMetrics();
-	    Display display = TiApplication.getAppRootOrCurrentActivity().getWindowManager().getDefaultDisplay();
-	    display.getMetrics(dm);
-	    int width = dm.widthPixels;
-	    int height = dm.heightPixels;
-	    return TiOrientationHelper.convertRotationToTiOrientationMode(display.getRotation(), width, height);
+		return  TiOrientationHelper.orientation;
```

Comment:
Change the `isPortrait()`, `isLandscape()`, `getPortrait()`, `getLandscape()`, and `getOrientation()` methods to use the new `TiOrientationHelper.getWindowTiOrientationModeFrom()` method I've suggested in my comments below. We want to calculate it everytime and not depend on external code to do it for us via globals.

You're definitely correct in removing usage of the `getConfiguration()` method call since that will fail to report orientation changes between "landscapeLeft" -> "landscapeRight" and vice-versa (this is by Google's design).

So, a call to `isPortrait()` should look like this...

```java
public boolean isPortrait()
{
	Activity activity = TiApplication.getAppRootOrCurrentActivity();
	int orientation = TiOrientationHelper.getWindowTiOrientationModeFrom(activity);
	return (orientation == ORIENTATION_PORTRAIT) || (orientation == ORIENTATION_PORTRAIT_REVERSE);
}
```

And a call to `isLandscape()` should look something like this...

```java
public boolean isLandscape()
{
	Activity activity = TiApplication.getAppRootOrCurrentActivity();
	int orientation = TiOrientationHelper.getWindowTiOrientationModeFrom(activity);
	return (orientation == ORIENTATION_LANDSCAPE) || (orientation == ORIENTATION_LANDSCAPE_REVERSE);
}
```